pkg/k2/constructs/template/property/construct_type.go

```diff
@@ -0,0 +1,69 @@
+package property
+
+import (
+	"fmt"
+	"github.com/klothoplatform/klotho/pkg/k2/model"
+	"gopkg.in/yaml.v3"
+	"strings"
+)
+
+type ConstructReference struct {
+	URN  model.URN `yaml:"urn" json:"urn"`
+	Path string    `yaml:"path" json:"path"`
+}
+
+type ConstructType struct {
+	Package string `yaml:"package"`
+	Name    string `yaml:"name"`
+}
+
+func (c *ConstructType) UnmarshalYAML(value *yaml.Node) error {
+	// Split the value into parts
+	parts := strings.Split(value.Value, ".")
```

Comment:
is `value.Value` safe to assume being the string? How does it behave if using a multiline string or other cases that have characters that wouldn't be present when unmarshalling to a string

nit: reduce the 3x copy of the parsing to one and reuse

Reply:
I don't understand what you mean by the nit. For the rest, I can decode the value to a string and validate it with a regex.

Reply:
Oh, the nit is that `UnmarshalYAML`, `ParseConstrucType` and `FromString` all do the same thing. Decoding to string is the most safe, since I know it'd behave well in those edge cases.